src/astro_databricks/operators/notebook.py

```diff
@@ -129,25 +131,58 @@ def _get_task_base_json(self) -> dict[str, Any]:
             "libraries": self.notebook_packages,
         }
 
+    def merge_notebook_packages(self, databricks_task_group: TaskGroup):
+        """
+        Merge the task group notebook packages into the notebook's packages, without adding any identical duplicates.
+        Modifies self.notebook_packages in place.
+
+        Example value for self.notebook_packages:
+        [
+            {"pypi": {"package": "requests_toolbelt==1.0.0"}}
+        ]
+
+        """
+        for task_group_package in databricks_task_group.notebook_packages:
+            exists = False
+            for existing_package in self.notebook_packages:
+                if task_group_package == existing_package:
+                    exists = True
+                    break
+            if not exists:
+                self.notebook_packages.append(task_group_package)
+
+    def find_parent_databricks_workflow_task_group(self):
+        """
+        Find the closest parent_group which is an instance of DatabricksWorkflowTaskGroup.
+
+        In the case of Airflow 2.2.4, inner Task Groups do not inherit properties from Parent Task Groups like more
+        recent versions of Airflow. This lead to the issue:
+        https://github.com/astronomer/astro-provider-databricks/pull/47
+        """
+        parent_group = self.task_group
+        while parent_group:
+            if parent_group.__class__.__name__ == "DatabricksWorkflowTaskGroup":
+                return parent_group
+            parent_group = parent_group._parent_group
+
     def convert_to_databricks_workflow_task(
         self, relevant_upstreams: list[BaseOperator], context: Context | None = None
     ):
         """
         Convert the operator to a Databricks workflow task that can be a task in a workflow
         """
-        if self.databricks_task_group and hasattr(
-            self.databricks_task_group,
-            "notebook_packages",
-        ):
-            self.notebook_packages.extend(self.databricks_task_group.notebook_packages)
-
-        if self.databricks_task_group and hasattr(
-            self.databricks_task_group,
-            "notebook_params",
-        ):
+        if airflow.__version__ == "2.2.4":
```

Comment:
should this be == 2.2.4 or <= 2.2.4?

Reply:
We only support from `>= 2.2.4`!  I just changed to take into account both (2.2.4 and 2.2.5).
 This could look nicer if we did the `packaging.version`, but then we'd introduce a new python dependency. WDYT, @jlaneve ?